src/util.c

```diff
@@ -600,3 +601,16 @@ mark_data_as_dirty(NetplanParser* npp, void* data_ptr)
         npp->current.netdef->_private->dirty_fields = g_hash_table_new(g_direct_hash, g_direct_equal);
     g_hash_table_insert(npp->current.netdef->_private->dirty_fields, data_ptr, data_ptr);
 }
+
+ssize_t
+netplan_copy_string(const char* input, char* out_buffer, size_t out_size)
```

Comment:
The function makes perfect sense.
praise: re-use of existing stpncpy is nice!

A small comment describing the input parameters and return value would be helpful here for future reference.

Reply:
This should be fixed in the latest revision.